mm/pmm.c

```diff
@@ -245,6 +245,39 @@ void init_pmm(void) {
     }
 }
 
+static frame_t *reserve_frame(frame_t *frame, unsigned int order) {
+    BUG_ON(!frame);
+    BUG_ON(!frame->free);
+    frame->free = false;
+
+    BUG_ON(frame->refcount > 0);
+    frame->refcount++;
+
+    list_unlink(&frame->list);
+    list_add(&frame->list, &busy_frames[order]);
+
+    return frame;
+}
+
+/* Reserves and returns the first free frame
+ * selected by the callback (returned true)
```

Comment:
nit: maybe fulfilling condition specified by the callback or something like this